qiskit/providers/aer/pulse_new/type_utils.py

```diff
@@ -13,46 +13,63 @@
 # that they have been altered from the originals.
 # pylint: disable=invalid-name
 
-"""Utilities for type handling/conversion for DE classes.
-
-A type specification is a dictionary describing a specific expected type, e.g. an array of a given
-shape. Currently only handled types are numpy arrays, specified via:
-    - {'type': 'array', 'shape': tuple}
+"""Utilities for type handling/conversion, primarily dealing with
+reshaping arrays, and handling qiskit types that wrap arrays.
 """
 
 import numpy as np
 
 
 class StateTypeConverter:
```

Comment:
I'm not sure I understand the purpose of this class. It seems to only be used in the denisty matrix problem / lindblad problem class, so could its functionality just be moved into that class?

Reply:
This class is also used in `DE_Methods.py` - I initially made it to automate vectorization in those classes as well. E.g. `solve_ivp` in `scipy` requires 1d arrays - the `ScipyODE(ODE_Method)` class contains an attribute that indicates this, and it automatically flattens the state of the de and wraps the inputs/outputs of the rhs function to work with the flattened version.

I was imagining this class to have utility as well when we start specifying initial states in terms of `terra.quantum_info` types. Maybe once the `BMDE_Solver` class is removed and we move more to the model of single-call solving (as opposed to a solver class you can keep interacting with) it will be less relevant.